cmake/farm_ng_add_protobufs.cmake

```diff
@@ -0,0 +1,61 @@
+macro(farm_ng_add_protobufs target)
+  set(multi_value_args NAMESPACE PROTO_FILES DEPENDENCIES)
+  cmake_parse_arguments(FARM_NG_ADD_PROTOBUFS "" "" "${multi_value_args}" ${ARGN})
+  set(${target}_PROTOBUF_IMPORT_DIRS ${CMAKE_CURRENT_SOURCE_DIR} CACHE STRING "Path to this project's protobuf sources")
+  foreach(_dep_target ${FARM_NG_ADD_PROTOBUFS_DEPENDENCIES})
+    list(APPEND DEP_PROTO_INCLUDES  -I ${${_dep_target}_PROTOBUF_IMPORT_DIRS})
+  endforeach()
+
+
+  if(NOT DEFINED FARM_NG_ADD_PROTOBUFS_NAMESPACE)
+    message(FATAL_ERROR "\nPlease specify NAMESPACE in farm_ng_add_protobufs(${target})\n")
+  endif()
+    
+  set(_proto_output_dir_cpp ${CMAKE_CURRENT_BINARY_DIR})
+  
+  # Extract the module name from the target
+  set(_cpp_out_sources)
+  set(_cpp_out_headers)
+  foreach (_proto_path ${FARM_NG_ADD_PROTOBUFS_PROTO_FILES})
+    SET(_full_proto_path ${CMAKE_CURRENT_SOURCE_DIR}/${_proto_path})
+    get_filename_component(_file_we ${_proto_path} NAME_WE)
+    get_filename_component(_file_dir ${_proto_path} DIRECTORY)
+
+
+    # cpp
+    set("_protoc_args_cpp"
+      "--cpp_out=${_proto_output_dir_cpp}")
+    SET(_cpp_out_src ${_proto_output_dir_cpp}/${_file_dir}/${_file_we}.pb.cc)
+    SET(_cpp_out_hdr ${_proto_output_dir_cpp}/${_file_dir}/${_file_we}.pb.h)
+    list(APPEND _cpp_out_all ${_cpp_out_src} ${_cpp_out_hdr})
+    list(APPEND _cpp_out_sources ${_cpp_out_src})
+    list(APPEND _cpp_out_headers ${_cpp_out_hdr})
+    add_custom_command(
+      OUTPUT ${_cpp_out_src} ${_cpp_out_hdr}
+      COMMAND ${PROTOBUF_PROTOC_EXECUTABLE}
+      ARGS ${_protoc_args_cpp} -I ${CMAKE_CURRENT_SOURCE_DIR} ${_full_proto_path}
+      DEPENDS ${_full_proto_path} ${PROTOBUF_PROTOC_EXECUTABLE}
+      COMMENT "Generating cpp protobuf code for ${_proto_path}"
+      VERBATIM)
+  endforeach()
+
+  farm_ng_add_library(${target}
+    NAMESPACE ${FARM_NG_ADD_PROTOBUFS_NAMESPACE}
+    INCLUDE_DIR ${_proto_output_dir_cpp}
+    HEADERS
+      ${_cpp_out_headers}
+    SOURCES
+      ${_cpp_out_sources})
+
+  target_link_libraries(${target} PUBLIC protobuf::libprotobuf ${FARM_NG_ADD_PROTOBUFS_DEPENDENCIES})
```

Comment:
This only worked for me on mac (Sophus CI) if 
${FARM_NG_ADD_PROTOBUFS_DEPENDENCIES} = lib::my_lib.

And I believe the following logic won't then be needed nor working correctly:

```
 foreach(_dep_target ${FARM_NG_ADD_PROTOBUFS_DEPENDENCIES})
    list(APPEND DEP_PROTO_INCLUDES  -I ${${_dep_target}_PROTOBUF_IMPORT_DIRS})
  endforeach()
  ```

Reply:
But I might be wrong. In the end all is good if/when CI passes here: https://github.com/strasdat/Sophus/pull/387